core/src/main/java/io/aiven/klaw/service/BaseOverviewService.java

```diff
@@ -224,9 +215,19 @@ protected List<AclOverviewInfo> applyFiltersAclsForSOT(
 
     List<AclOverviewInfo> aclList = new ArrayList<>();
     AclOverviewInfo mp;
-
+    Set<Integer> set =
+        aclRequestsRepo
+            .findAllByTenantIdAndEnvironmentAndRequestStatusAndTopicname(
+                tenantId,
+                aclsFromSOT.get(0).getEnvironment(),
+                RequestStatus.DELETED.value,
+                aclsFromSOT.get(0).getTopicname())
+            .stream()
+            .map(AclRequests::otherParams)
+            .collect(Collectors.toSet());
     for (Acl aclSotItem : aclsFromSOT) {
       if (aclSotItem.getAclip() != null || aclSotItem.getAclssl() != null) {
+        boolean deleteRequestExists = set.contains(aclSotItem.getReq_no());
         mp = new AclOverviewInfo();
         mp.setEnvironment(aclSotItem.getEnvironment());
```

Comment:
Another requirement will be to update TopicOverviewServiceTest.java 
The addition of aclRequestsRepo in this path means it needs to be added as a mock 
e.g. 
@Mock
  protected AclRequestsRepo aclRequestsRepo;
  
  then in the setUp Method use ReflectionTestUtils to assign it to the topicOverService
  
  after that you will need to setup a when clause to return a response when it is called.

Ideally this should have at least one test where it does return an open delete request so that it can be tested that it returns the correct boolean.

Aisde from that I think it looks good!